README.md

```diff
@@ -1,40 +1,92 @@
-* Vscode: <https://code.visualstudio.com/Download>
-* Vscode extensions for Rust:
-  * crates <https://marketplace.visualstudio.com/items?itemName=serayuzgur.crates>
-  * Rust <https://marketplace.visualstudio.com/items?itemName=rust-lang.rust>
-  * Even Better TOML <https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml>
-  * Power Header <https://marketplace.visualstudio.com/items?itemName=epivision.vscode-file-header>
-    (Enables you to insert a copyright header in VSCode by pressing CTRL+ALT+H)
-  * Rust Test Explorer <https://marketplace.visualstudio.com/items?itemName=swellaby.vscode-rust-test-adapter>
-  * Rust debugger CodeLLDB <https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb>
-  * Coverage Gutters <https://marketplace.visualstudio.com/items?itemName=ryanluker.vscode-coverage-gutters>
+DBSP intends to change the situation by offering a simple promise: to **execute
+any query workload that can run in batch mode in streaming mode**.
 
-## Check test coverage
+## Overview
 
-* Run on the command line `cargo tarpaulin --out Lcov` to generate coverage information
-* Use the Coverage Gutters VSCode extension to watch the lcov file (CTRL+SHIFT+P, Coverage Gutters: Watch)
+DBSP queries are composed of the following classes of operators (in the order of
+increasing complexity):
 
-## Set-up git hooks
+1. **Per-record operators** that parse, validate, filter, transform data streams
+one record at a time.
+
+1. **Windowing operators** that group time series data into time windows,
+including various forms of tumbling and sliding windows, windows driven by
+watermarks, etc.
+
+1. The complete set of **relational operators**: select, project, join,
+aggregate, etc.
+
+1. **Recursion**: Recursive queries express iterative computations, e.g.,
+partitioning a graph into strongly connected components.  Like all DPSP queries,
+recursive queries update their outputs incrementally as new data arrives.
+
+DBSP unifies two kinds of streaming data: time series data and change data.
+
+- **Time series data** can be thought of as an infinitely growing log indexed by
+  time.
+
+- **Change data** represents updates (insertions, deletions, modifications) to
+  some state modeled as a table of records.
+
+In DBSP, a time series is just a table where records are only ever added and
+never removed or modified.  As a result, this table can grow unboundedly; hence
+most queries work with subsets of the table within a bounded time window.  DBSP
+does not need to wait for all data within a window to become available before
+evaluating a query (although the user may choose to do so): like all queries,
+time window queries are updated on the fly as new inputs become available.  This
+means that DBSP can work with arbitrarily large windows as long as they fit
+within available storage.  All other operators listed above apply to both time
+series and change data.
```

Comment:
Perhaps I could take a stab at improving this documentation, after all I have already tons of slides I could steal material from. But I understand that this text should be more developer-oriented.